### PR TITLE
Add index mapping parameter for counted_keyword

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we add a new mapping parameter `index` to the `counted_keyword` mapping type. This allows to reduce disk usage for use cases where indexed fields are not required.

Relates #101826

### CODE REVIEW

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/counted_keyword/10_basic.yml [hunk 1]
```diff
@@ -49,3 +66,20 @@ setup:
   - match: { aggregations.event_terms.buckets.2.key: "c" }
   - match: { aggregations.event_terms.buckets.2.doc_count: 2 }
   - length: { aggregations.event_terms.buckets: 3 }
+
+# although the field is not indexed, the counted_terms agg should still work
+  - do:
```
Comment (martijnvg) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/counted_keyword/10_basic.yml:71]: Maybe also verify that no inverted index is created for the `events` field in `test-event-no-index` index? Maybe by executing a field caps api call and/or disk usage api call and verify the response?

File: x-pack/plugin/mapper-counted-keyword/src/main/java/org/elasticsearch/xpack/countedkeyword/CountedKeywordFieldMapper.java [hunk 2]
```diff
@@ -78,16 +78,24 @@ public class CountedKeywordFieldMapper extends FieldMapper {
     public static final String CONTENT_TYPE = "counted_keyword";
     public static final String COUNT_FIELD_NAME_SUFFIX = "_count";
 
-    public static final FieldType FIELD_TYPE;
+    private static final FieldType FIELD_TYPE_INDEXED;
```
Comment (martijnvg) [inline x-pack/plugin/mapper-counted-keyword/src/main/java/org/elasticsearch/xpack/countedkeyword/CountedKeywordFieldMapper.java:81]: Maybe also update the `CountedKeywordFieldMapperTests` test case to have a test similar to `KeywordFieldMapperTests$testDisableIndex()`?

**Review by martijnvg — COMMENTED** [review]
Left two testing comments. Looks good otherwise.

File: x-pack/plugin/mapper-counted-keyword/src/main/java/org/elasticsearch/xpack/countedkeyword/CountedKeywordFieldMapper.java (hunk 2 above)
Comment (danielmitterdorfer) [reply x-pack/plugin/mapper-counted-keyword/src/main/java/org/elasticsearch/xpack/countedkeyword/CountedKeywordFieldMapper.java:81]: Added in https://github.com/elastic/elasticsearch/pull/103646/commits/47732ce0b522f2c6b9332033dc6f905ca478ad06.

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/counted_keyword/10_basic.yml (hunk 1 above)
Comment (danielmitterdorfer) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/counted_keyword/10_basic.yml]: Good point! I've also split the test case files for the default options and for `index: false` as the latter will be introduced in 8.13 + it keeps the two cases neatly separated.

**Review by martijnvg — APPROVED** [review]
LGTM

**Pull request merged**